src/server/graph-generation/cytoscapeJson.js

```diff
@@ -8,34 +8,28 @@ const metadataMapperPC2 = require('./metadataMapperPC2');
 function getPathwayLevelMetadata(uri) {
   let title, dataSource, comments, organism;
 
-  //Get title
-  return pcServices.traversePC2(uri, 'Named/displayName').then(function (data) 
-{  title = data.traverseEntry[0].value;
-
-    //Get data source
-    return pcServices.traversePC2(uri, 'Entity/dataSource/displayName').then(function (data) {
+  return (Promise.resolve(pcServices.traversePC2(uri, 'Named/displayName'))
+    .then(data => {
+      title = data.traverseEntry[0].value;
+      return pcServices.traversePC2(uri, 'Entity/dataSource/displayName');
+    })
+    .then(data => {
       dataSource = data.traverseEntry[0].value;
-
-      //Get comments
-      return pcServices.traversePC2(uri, 'Entity/comment').then(function (data) {
-        comments = data.traverseEntry[0].value;
-
-        //Get organism name
-        return pcServices.traversePC2(uri, 'Entity/organism/displayName').then(function (data) {
-          organism = data.traverseEntry[0].value;
-
-          //Return pathway metadata
-          return {
-            comments: comments,
-            dataSource: dataSource,
-            title: title,
-            organism: organism
-          };
-
-        })
-      });
-    });
-  });
+      return pcServices.traversePC2(uri, 'Entity/comment');
+    })
+    .then(data => {
+      comments = data.traverseEntry[0].value;
+      return pcServices.traversePC2(uri, 'Entity/organism/displayName');
+    })
+    .then(data => {
+      organism = data.traverseEntry[0].value;
+      return {
+        comments,
+        dataSource,
+        title,
+        organism
+      }
```

Comment:
Much better!  The next step to make things even faster and easier to read is to use short, arrow functions.

Something like:

```js
let getValue = data => data.where.the.field.is;
let get = name => getFromPc( name ).then( getValue );
```

Then you do a bunch in parallel:

```js
let foo, bar;

Promise.all([
  get('x/y').then( v => foo = v ),
  get('a/b').then( v => bar = v )
]);
```

If you have N things that you're getting, you can be up to N times faster this way, and it's more clear and concise.